### DISCUSSION TITLE
Accessing the current LDAP user in a task

### DISCUSSION THREAD

**Discussion by GMTurbo** [apache/airflow#30931] [Q&A] [answered]
I have an instance of airflow running that is using LDAP.  All that is working correctly, but i'm having issues accessing the current ldap user in a virtualenv task.  My workflow writes to external tables and with multiple users in our LDAP group, I want to be able to log who was running what.

**Comment by potiuk** [comment, marked as answer]
You can query the event log for informatio about "triggered" dag https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#tag/EventLog. There is no such infomration in the actual run because in general airflow tasks are scheduled automatically.

You can at most see the context of execution from withing the venv (read the docstrings of venv, they wil explain you the conditions when the context is available - Python and Airflow Version must be the same in venv as in airflow in general)  https://github.com/apache/airflow/blob/main/airflow/utils/context.pyi

**Reply by GMTurbo** [reply]
i'll try accessing the event log.  Thanks.  I'll report back if that works for me.

**Reply by GMTurbo** [reply]
So i was able to access the Log table with execution_date, dag_id, and run_id to query the user responsible for trigger and clear events.
